modules/fedimint-ln-client/src/lib.rs

```diff
@@ -1279,49 +1221,65 @@ impl LightningClientModule {
+                                if !change.is_empty() {
+                                    yield LnPayState::AwaitingChange;
+                                    match client_ctx.await_primary_module_outputs(operation_id, change.clone()).await {
+                                        Ok(_) => {
+                                            yield LnPayState::Success { preimage };
+                                        }
+                                        Err(e) => {
+                                            yield LnPayState::UnexpectedError {
+                                                error_message: format!("Error occurred while waiting for the change: {e:?}"),
+                                            };
+                                        }
+                                    }
+                                } else {
+                                    yield LnPayState::Success { preimage };
                                 }
+                                return;
                             }
-                        }
-
-                        yield LnPayState::Success {preimage};
-                        return;
-                    }
-                    Err(PayError::Refundable(block_height, error)) => {
-                        yield LnPayState::WaitingForRefund{ block_height, gateway_error: error.clone() };
+                            LightningPayStates::Refund(refund) => {
+                                yield LnPayState::WaitingForRefund {
+                                    error_reason: refund.error_reason.clone(),
+                                };
+
+                                match client_ctx.await_primary_module_outputs(operation_id, refund.out_points).await {
+                                    Ok(_) => {
+                                        yield LnPayState::Refunded {
+                                            error_reason: refund.error_reason,
+                                        };
+                                    }
+                                    Err(e) => {
+                                        yield LnPayState::UnexpectedError {
+                                            error_message: format!("Error occurred trying to get refund. Refund was not successful: {e:?}"),
+                                        };
+                                    }
+                                }
 
-                        if let Ok(out_points) = self_ref.await_refund(operation_id).await {
-                            // need to await primary module to get refund
-                            if client_ctx.await_primary_module_outputs(operation_id, out_points).await.is_ok() {
-                                yield LnPayState::Refunded { gateway_error: error };
                                 return;
                             }
+                            _ => {}
                         }
+                        Some(_) => {}
+                        None => {}
```

Comment:
nit: should debug-log that stream closed when receiving `None`. This isn't necessarily unexpected on shutdown, but if it happens during normal operation it could indicate a problem.